File: ruby/ExampleLoginWebViewTest.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'spec'
require 'appium_lib'
require 'selenium-webdriver'

describe 'Login' do

	EMAIL_TEST = 'test_user@example.com'
	PASSWORD_TEST = 'test_PASSWORD_1234'
	RELATIVE_APK_PATH = '../relative/path/to/your/application.apk'
	PLATFORM_VERSION = '4.4'
	EMULATOR_NAME = 'emulator_name'
	WEBVIEW_CONTEXT = 'WEBVIEW_com.example.main'
	APPIUM_VERSION = '1.4.11'

	def desired_caps
		{
			caps: {
				:'appium-version' => APPIUM_VERSION,
				platformName: 'Android',
				platformVersion: PLATFORM_VERSION,
				deviceName: EMULATOR_NAME,
				app: RELATIVE_APK_PATH,
				name: 'Sample mobile test'
			},
			appium_lib: {
				wait: 360
			}
		}
	end

	before do
		Appium::Driver.new(desired_caps).start_driver
		set_context(WEBVIEW_CONTEXT)
	end

	after do
		driver_quit
	end

	it 'test user can login and go to home' do
		# Given:
		email_input = find_element(:id,'emailInputId')
		email_input.clear()
		password_input = find_element(:id,'passwordInputId')
		password_input.clear()
		submit_button = find_element(:id,'loginButtonId')

		# When:
		email_input.type(EMAIL_TEST)
		password_input.type(PASSWORD_TEST)
		submit_button.click()

		# Then:
		sleep(5)
		otherNewsHome = driver.find_element(:id,'anyResultPageContentId')
		otherNewsHome.instance_of?Selenium::WebDriver::Element
	end
end

print Minitest.run_specs({ :info => [__FILE__] }).first

File: python/ExampleLoginWebViewTest.py
#!/usr/bin/env python

import os
import glob
import unittest
from time import sleep
from appium import webdriver

class AndroidWebViewTests(unittest.TestCase):

	EMAIL_TEST = 'test_user@example.com'
	PASSWORD_TEST = 'test_PASSWORD_1234'
	PLATFORM_VERSION = '4.4'
	RELATIVE_APK_PATH = '../relative/path/to/your/application.apk'
	MAIN_ACTIVITY = '.MainActivity'
	PACKAGE = 'com.example.main'
	EMULATOR_NAME = 'emulator_name'
	APPIUM_SERVER = 'http://localhost:4723/wd/hub'
	WEBVIEW_CONTEXT = 'WEBVIEW_com.example.main'

	def setUp(self):
		app = os.path.abspath(
				os.path.join(os.path.dirname(__file__),
					RELATIVE_APK_PATH))

		desired_caps = {
			'app': app,
			'appPackage': PACKAGE,
			'appActivity': MAIN_ACTIVITY,
			'platformName': 'Android',
			'platformVersion': PLATFORM_VERSION,
			'deviceName': EMULATOR_NAME
		}

		if (PLATFORM_VERSION != '4.4'):
			desired_caps['automationName'] = 'selendroid'

		self.driver = webdriver.Remote(APPIUM_SERVER,
			desired_caps)
		self.driver.switch_to.context(WEBVIEW_CONTEXT)

	def test_user_can_login_and_go_to_home(self):
		# Given:
		email_input = self.driver.find_element_by_id('emailInputId')
		email_input.clear()
		password_input = self.driver.find_element_by_id('passwordInputId')
		password_input.clear()
		submit_button = self.driver.find_element_by_id('loginButtonId')

		# When:
		email_input.send_keys(EMAIL_TEST)
		password_input.send_keys(PASSWORD_TEST)
		submit_button.click()

		# Then:
		sleep(5)
		source = self.driver.page_source
		self.assertNotEqual(-1, source.find("anyResultPageContent"))

	def tearDown(self):
		self.driver.quit()


if __name__ == '__main__':
	suite = unittest.TestLoader().loadTestsFromTestCase(AndroidWebViewTests)
	unittest.TextTestRunner(verbosity=2).run(suite)
